src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const changedState = state;
+
+  for (const action of Object.values(actions)) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const [addKey, addValue] of Object.entries(action.extraData)) {
+          changedState[addKey] = addValue;
+        }
+        break;
+
+      case 'removeProperties':
+        for (const removeKey of Object.values(action.keysToRemove)) {
```

Comment:
`keysToRemove` is an array so you use directly:
```suggestion
        for (const removeKey of action.keysToRemove) {
```